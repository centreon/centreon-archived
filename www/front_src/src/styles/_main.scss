@import "partials/typography";
@import "partials/fonts";
@import "partials/variables";
@import "partials/messages";
@import "partials/icons";
@import "partials/form/form";
@import "partials/helpers";
@import "partials/mixins";
