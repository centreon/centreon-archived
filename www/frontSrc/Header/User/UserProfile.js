import React from 'react'
import { withStyles } from 'material-ui/styles'
import Avatar from 'material-ui/Avatar'
import { MenuItem } from 'material-ui/Menu'
import IconButton from 'material-ui/IconButton'
import Grid from 'material-ui/Grid'
import Popover from 'material-ui/Popover'
import Button from 'material-ui/Button'
import Bookmark from 'material-ui-icons/Bookmark'
import VolumeUp from 'material-ui-icons/VolumeUp'
import Typography from 'material-ui/Typography'
import Moment from 'moment'

const styles = theme => ({
  profileRoot: {
    display: 'flex',
    flexDirection: 'row-reverse'
  },
  moment: {
    alignSelf: 'flex-start',
    margin: '10px 0px',
  },
  headerDate: {
    fontSize: '1.5vw',
    lineHeight: '16px',
    margin: '2px 0',
  },
  headerTime: {
    fontSize: '2.5vw',
    lineHeight: '20px',
  },
  avatarButton: {
    alignSelf: 'flex-end',
    margin: '6px 4px'
  },
  avatar: {
    backgroundColor: '#FDFEFE',
    width: 36,
    height: 36,
    display: 'inline-flex',
    verticalAlign: 'middle',
    color: '#24323E',
  },
  profile: {
    padding: '10px 16px',
    borderBottom: '1px solid #d1d2d4',
  },
  profileLink: {
    float: 'right'
  },
  logoutButton: {
    textTransform: 'initial',
    border: '1px solid ' + theme.palette.error.main,
    color: theme.palette.error.main,
  },
  menuFooter: {
    display: 'flex',
    flexDirection: 'row-reverse',
    padding: '10px 16px',
    boxSizing: 'content-box',
    borderTop: '1px solid #d1d2d4',
  }
})

const UserProfile = ({
    classes,
    open,
    handleOpen,
    handleClose,
    anchorEl
  }) => (
  <Grid item xs={3}>
    <div className={classes.profileRoot}>
      <IconButton
        aria-haspopup="true"
        onClick={handleOpen}
        className={classes.avatarButton}
      >
        <Avatar className={classes.avatar}>
          RI
        </Avatar>
      </IconButton>
      <div className={classes.moment}>
        <div className={classes.headerDate}> {Moment(new Date()).format('LL')} </div>
        <div className={classes.headerTime}> {Moment().format('LT')} </div>
      </div>
    </div>
    <Popover
      open={open}
      anchorEl={anchorEl}
      anchorReference='anchorEl'
      anchorPosition={{ top: 200, left: 400 }}
      onClose={handleClose}
      anchorOrigin={{
        vertical: 'bottom',
        horizontal: 'right',
      }}
      transformOrigin={{
        vertical: 'top',
        horizontal: 'right',
      }}
    >
      <div className={classes.profile}>
        <Typography component="title" gutterBottom>
          Rabaa Ridene
        </Typography>
        <Typography variant="caption" gutterBottom>
          as Rayden <a href="./main.php?p=50104&o=c" className={classes.profileLink}>Edit profile </a>
        </Typography>
      </div>
      <MenuItem onClick={handleClose}>
        <Bookmark /> Add to your bookmark
      </MenuItem>
      <MenuItem onClick={handleClose}>
        <VolumeUp /> DÃ©sactivate notification sonore
      </MenuItem>
      <div className={classes.menuFooter}>
        <a href="index.php?disconnect=1">
          <Button className={classes.logoutButton}>
            Sign out
          </Button>
        </a>
      </div>
    </Popover>
  </Grid>
    )

export default withStyles(styles)(UserProfile)
