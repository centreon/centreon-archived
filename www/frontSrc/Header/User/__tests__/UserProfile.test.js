/* eslint-env jest */
import React from 'react'
import Enzyme from 'enzyme'
import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16'
import UserProfile from '../UserProfileContainer'
import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import { MuiThemeProvider, createMuiTheme } from '@material-ui/core/styles'

Enzyme.configure({ adapter: new Adapter() })

const theme = createMuiTheme({
  font: {
    openSans: "'Open Sans', Arial, Tahoma, Helvetica, Sans-Serif"
  },
})

describe('UserProfile', () => {
  const middlewares = [thunk]
  const mockStore = configureStore(middlewares)

  const store = mockStore({
    user: {
      data: {
        autologinkey: "f855810b7eafb9cfb0c3d74c62af0fb2e2647939",
        fullname: "admin",
        hasAccessToProfile: false,
        locale: "en_US",
        soundNotificationsEnabled: false,
        timezone: "65",
        username: "admin"
      }
    }
  })

  let getUser = jest.fn()

  const component = mount(
    <MuiThemeProvider theme={theme}>
      <UserProfile store={store} getUser={getUser} />
    </MuiThemeProvider>
  )

  it('User Component renders correctly', () => {
    const userProfile = shallow(<UserProfile store={store} />)
    expect(toJson(userProfile)).toMatchSnapshot()
  })

  it('should display popover when clicking on user icon', () => {

    expect(component.find('div#userPopover').length).toEqual(0)
    component.find('button#userIcon').simulate('click')
    expect(component.find('div#userPopover').length).toEqual(1)
  })

  it('should handle Autologin', () => {
    component.find('li#autologinAction').simulate('click')

    const actions = store.getActions()
    expect(actions[2].type).toBe('REQUEST_AUTOLOGIN')
  })

  it('should handle notification sound', () => {
    component.find('li#notifAction').simulate('click')

    const actions = store.getActions()
    expect(actions[4].type).toBe('REQUEST_ENABLED_NOTIF')
  })
})