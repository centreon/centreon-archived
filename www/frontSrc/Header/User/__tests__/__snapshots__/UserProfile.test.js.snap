// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfile User Component renders correctly 1`] = `
<UserProfileContainer
  autoLogin={[Function]}
  getUser={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
  user={
    Object {
      "autologinkey": "f855810b7eafb9cfb0c3d74c62af0fb2e2647939",
      "fullname": "admin",
      "hasAccessToProfile": false,
      "locale": "en_US",
      "soundNotificationsEnabled": false,
      "timezone": "65",
      "username": "admin",
    }
  }
/>
`;
